main.tf

```diff
@@ -0,0 +1,70 @@
+###################################################################
+# Instance module usage
+###################################################################
+
+data "ibm_resource_group" "resource_group" {
+  name = (var.resource_group != null ? var.resource_group : "default")
+}
+
+module "sshkey" {
+  source = "./modules/ssh-key"
+
+  for_each          = { for key in var.key_list : key.name => key }
+  is_provision      = true
+  name              = each.value.name
+  resource_group_id = data.ibm_resource_group.resource_group.id
+  public_key        = each.value.public_key
+  tags              = each.value.tags
+}
+
+module "storage" {
+  source = "./modules/storage"
+
+  for_each          = { for volume in var.volume_list : volume.name => volume }
+  is_provision      = true
+  name              = each.value.name
+  resource_group_id = data.ibm_resource_group.resource_group.id
+  location          = each.value.location
+  volume_profile    = each.value.volume_profile
+  iops              = each.value.iops
+  size              = each.value.size
+  encryption        = each.value.encryption
+  tags              = each.value.tags
+}
+
+locals {
+  sshkeys = [for key in module.sshkey : key.ssh_key_id]
+  volumes = [for volume in module.storage : volume.volume_id]
+}
+
+module "instance" {
```

Comment:
Please check missing arguments  like
default_trusted_profile_auto_link
crn
port_speed

make sure we cover all mandatory and optional arguments

Reply:
yes. have covered all of the parameters. These might have been added very recently after the developement phase. will be addressing them in the upcoming PR